src/Quartz.Serialization.SystemTextJson/Converters/CalendarConverter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+
+using Quartz.Calendars;
+using Quartz.Impl.Calendar;
+using Quartz.Util;
+
+namespace Quartz.Converters;
+
+internal class CalendarConverter : JsonConverter<ICalendar>
+{
+    private static readonly Dictionary<string, ICalendarSerializer> converters = new()
+    {
+        { BaseCalendarSerializer.CalendarTypeKey, new BaseCalendarSerializer() },
```

Comment:
nit: for thread-safe types could probably also have `BaseCalendarSerializer.Instance`, but the dictionary already is static so does not matter that much

Reply:
Was thinking about that also but didn't implement for some reason. I will add the instance methods.